### ISSUE TITLE
Questions in example of Modeling solubility

### ISSUE THREAD

**Issue by sym823808458** [deepchem/deepchem#1741]
Description:

DeepChem Version:deepchem 2.10
Platform:linux Ubuntu
I run the example Modeling solubility,and feel doubtful about the some concepts.
1.what does that means "train_score"?The closer you get to 1, the better?
2.The code works well ,but the results are terrible.How can I optmize them as much as possible?maybe fitting models for several times?how could I do that?
here is my code.I upload my own dataset in this framework.

```
import numpy.random

params_dict = {"learning_rate": np.power(10., np.random.uniform(-5, -3, size=1)),
               "decay": np.power(10, np.random.uniform(-6, -4, size=1)),
               "nb_epoch": [20] }
n_features = train_dataset.get_data_shape()[0]
def model_builder(model_params, model_dir):
  model = dc.models.MultitaskRegressor(
    1, n_features, layer_sizes=[1000], dropouts=[.25],
    batch_size=50, **model_params)
  return model

optimizer = dc.hyper.HyperparamOpt(model_builder)
best_dnn, best_dnn_hyperparams, all_dnn_results = optimizer.hyperparam_search(
    params_dict, train_dataset, valid_dataset, transformers,
    metric=metric)
```
here are the results.
Fitting model 1/1
hyperparameters: {'learning_rate': 0.00013651785919268744, 'decay': 2.476133299510563e-05, 'nb_epoch': 20}
computed_metrics: [-0.2507983931835007]
Model 1/1, Metric r2_score, Validation set 0: -0.250798
	best_validation_score so far: -0.250798
computed_metrics: [-0.058752113559920494]
Best hyperparameters: (0.00013651785919268744, 2.476133299510563e-05, 20)
train_score: -0.058752
validation_score: -0.250798

Thanks a lot

**Comment by rbharath** [comment]
1. Train score is the results of your metric on training. Usually this should be close to 1 if your model is training well (but not too close in which case you're overfitting)
2. I'm not sure what's going on here since I don't know what dataset you're working with. I'd recommend working through the deepchem tutorials to start with. Can you get good results on them? I'd next recommend setting up a baseline, perhaps with a simple machine learning model such as a random forest. Then once you have a baseline set up, try a model like the multitask regressor.

**Comment by sym823808458** [comment]
Thanks a lot for your response!
Finally I figured it out that the terrible results were caused by the spliter,because in my case,scaffold method will cause big difference.Anyway,the train score could be 0.90 and R2 reached 0.35 by random splitter.
However,I wanna figure it out that which featurization method could be better for my case.After reading the paper ( Chem. Sci., 2018, 9, 513),I am confused that which feat method is coresponded with ECFP?maybe the CircularFigerprint I used?Could you please tell me?
More details about the feat method are needed.Could you give me any resources?
Thanks for your friendly help.

**Comment by rbharath** [comment]
Yes, the CircularFingerprint is ECFP. 

For more resources, I'd definitely recommend working through the tutorials. They're structured to teach the material and not just the API.

**Comment by sym823808458** [comment]
Thank you for everything you give.It helps a lot.
